src/lib/metrics_extractor/metrics_extractor_utils.py

```diff
@@ -109,3 +110,26 @@ def get_last_update_time(
             return datettime_utc
 
     return None
+
+
+def get_min_update_time(
```

Comment:
Function name might become a bit misleading in the future, I guess.
As it searches for the min_update_time AMONG last_update_time.

So, maybe get_earliest_last_update_time or get_earliest_last_update_time_for_resource_set might be a bit more descriptive. What do you think?

Reply:
yes, totally agree with your point